src/Exception/WaiterException.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrCash\Exception;
+
+/**
+ * In some cases, Stripe sends a lot of different events nearly simultaneously. For example, when creating
+ * a new subscription, Stripe automatically generates both an "invoice.created" and "invoice.payment_succeeded"
+ * event. However, when processed server-side, there must be some order that must be guaranteed.
+ *
+ * As Stripe resends events whenever it receives a non-200 HTTP status code, we can throw this exception to
+ * force a non-200 error, and let Stripe resends the event later
+ *
+ * @author Michaël Gallego <mic.gallego@gmail.com>
+ */
+class WaiterException extends RuntimeException implements ExceptionInterface
```

Comment:
Consider using `public static` constructors to make the code more readable where the exceptions are used